### ISSUE TITLE
Unable to find current origin/monero revision in submodule path 'external/miniupnp'

### ISSUE THREAD

**Issue by yodel-settles** [aeonix/aeon#266]
I run **git submodule update --init --force --remote --recursive**

fatal: Needed a single revision
Unable to find current origin/monero revision in submodule path 'external/miniupnp'

**Comment by iamsmooth** [comment]
I think this is ubuntu 20, according to the comment on bitcointalk

**Comment by stoffu** [comment]
Just remove --remote flag.

**Comment by iamsmooth** [comment]
@yodel-settles Have you tried the build as described in the readme? Those options aren't there.

**Comment by yodel-settles** [comment]
@iamsmooth yes, already tried instructions as written in readme ... after those failed, tried more combos ... tried about 5 different combos none worked...

also aeon does not build on 18.04 either just spun up VM and it has same issue...which is why I tried Ubuntu 20 in the first place tbh...

**Comment by yodel-settles** [comment]
@stoffu not that simple, sorry. just tried with Ubuntu 18.04 VM

using vanilla build instructions on Ubuntu 18.04, which is same exact issue as Ubuntu 20...

-- CMake version 3.10.2
-- The C compiler identification is GNU 7.5.0
-- The CXX compiler identification is GNU 7.5.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Building without build tag
-- Found Git: /usr/bin/git (found version "2.17.1")
-- Checking submodules
CMake Error at CMakeLists.txt:185 (message):
Submodule 'external/miniupnp' is not up-to-date. Please update with

git submodule update --init --force external/miniupnp

or run cmake with -DMANUAL_SUBMODULES=1
Call Stack (most recent call first):
CMakeLists.txt:190 (check_submodule)

-- Configuring incomplete, errors occurred!
See also "aeon/build/Linux/HEAD_detached_at_v0.14.1.0-aeon/release/CMakeFiles/CMakeOutput.log".
Makefile:94: recipe for target 'release-all' failed
make: *** [release-all] Error 1
aeon# git submodule update --init --force external/miniupnp
fatal: remote error: upload-pack: not our ref 4c700e09526a7d546394e85628c57e9490feefa0
fatal: The remote end hung up unexpectedly
Fetched in submodule path 'external/miniupnp', but it did not contain 4c700e09526a7d546394e85628c57e9490feefa0. Direct fetching of that commit failed.

**Comment by yodel-settles** [comment]
using vanilla build instructions on Ubuntu 18.04, which is same exact issue as Ubuntu 20...

-- CMake version 3.10.2
-- The C compiler identification is GNU 7.5.0
-- The CXX compiler identification is GNU 7.5.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Building without build tag
-- Found Git: /usr/bin/git (found version "2.17.1")
-- Checking submodules
CMake Error at CMakeLists.txt:185 (message):
  Submodule 'external/miniupnp' is not up-to-date.  Please update with

  git submodule update --init --force external/miniupnp

  or run cmake with -DMANUAL_SUBMODULES=1
Call Stack (most recent call first):
  CMakeLists.txt:190 (check_submodule)


-- Configuring incomplete, errors occurred!
See also "aeon/build/Linux/_HEAD_detached_at_v0.14.1.0-aeon_/release/CMakeFiles/CMakeOutput.log".
Makefile:94: recipe for target 'release-all' failed
make: *** [release-all] Error 1
aeon# git submodule update --init --force external/miniupnp
fatal: remote error: upload-pack: not our ref 4c700e09526a7d546394e85628c57e9490feefa0
fatal: The remote end hung up unexpectedly
Fetched in submodule path 'external/miniupnp', but it did not contain 4c700e09526a7d546394e85628c57e9490feefa0. Direct fetching of that commit failed.

**Comment by stoffu** [comment]
@yodel-settles 
You seem to be building against some commit which is not the current latest master. The commit hash `4c700e09526a7d546394e85628c57e9490feefa0` of miniupnp was changed in this patch https://github.com/aeonix/aeon/commit/95388f84bad7157879e8641117d035b71b9c42e8.

Delete entirely the current repository directory, and do
```
git clone --recursive https://github.com/aeonix/aeon
cd aeon
make
```

The code builds fine on both Ubuntu 18 & 20. That's what we ensure through GitHub Actions CI.

**Comment by stoffu** [comment]
Now I understand what's going on: the miniupnp submodule's URL was changed from https://github.com/monero-project/miniupnp/ to https://github.com/miniupnp/miniupnp in this patch https://github.com/aeonix/aeon/commit/95388f84bad7157879e8641117d035b71b9c42e8 which was merged after the `v0.14.1.0-aeon` release. When you do `git clone --recursive https://github.com/aeonix/aeon`, it checks out the current master where the miniupnp submodule points to `544e6f` which belongs to the new URL, but then when you check out `v0.14.1.0-aeon`, the submodule now points to `4c700e` which belongs to the old URL and doesn't exist in the new URL. That's why this CMake error message was caused.

To get around this error, just skip checking out to the `v0.14.1.0-aeon` release, and build the current master:
```
git clone --recursive https://github.com/aeonix/aeon
cd aeon
make
```

This frustrating issue is rather which happens only when one changes a submodule's URL. The next point release `v0.14.2.0-aeon` is coming very soon, so I think we can leave the README as is meanwhile.

**Comment by yodel-settles** [comment]
looks like it works now, thanks
